script.js

```diff
@@ -0,0 +1,364 @@
+      saveData(first);
+    }
+  }
+
+  function addOperator(operatorEvent) {
+    // Evaluate expression when chaining operations with a new sign
+    if (first.saved && operator.saved && second.saved) {
+      evaluateExpression(first, operator, second);
+    }
+    if (operatorEvent === "-" && operator.saved && second.unsaved) {
+      // Allow a leading minus sign for the second number
+      second.value = operatorEvent;
+      saveData(second);
+      // Handle both negative sign and number
+      if (operator.value === "-" && second.value.startsWith("-")) {
+        operator.value = "+";
+        resetData(second);
+      }
+      // Handle a plus sign and followed by a negative second number
+      if (operator.value === "+" && second.value.startsWith("-")) {
+        operator.value = "-";
+        second.value = second.value.substring(1);
+        resetData(second);
+      }
+    }
+    // Save the sign for the expression
+    if (first.saved && operator.unsaved && first.value !== "-") {
+      operator.value = operatorEvent;
+      saveData(operator);
+    }
+    // Allow a leading minus sign for the first number
+    if (operatorEvent === "-" && first.unsaved && operator.unsaved) {
+      first.value = operatorEvent;
+      saveData(first);
+    }
+  }
+
+  function evaluateExpression(first, operator, second) {
+    // Handle a complete expression
+    if (first.saved && operator.saved && second.saved) {
+      // Handle a single minus sign as a second number, like 25x-
+      if (second.value === "-") {
+        return;
+      }
+      result = operate(first.value, operator.value, second.value);
+    }
+    // Handle a single first number followed by a sign
+    if (first.saved && operator.saved && second.unsaved) {
+      second.value = first.value;
+      result = operate(first.value, operator.value, second.value);
+    }
+    // Handle a single first number
+    if (first.saved && operator.unsaved && second.unsaved) {
+      result = first.value;
+    }
+    // Handle an empty expression
+    if (first.unsaved && operator.unsaved && second.unsaved) {
+      result = "0";
+    }
+    // Allow to chain expression when adding a sign
+    if (result !== "Error") {
+      first.value = result;
+      allowNewNumber = true;
+      [operator, second].forEach(element => resetData(element));
+    } else {
+      [first, operator, second].forEach(element => resetData(element));
+    }
+  }
+
+  function operate(first, sign, second) {
+    const a = Number(first);
+    const b = Number(second);
+    switch (sign) {
+      case "+":
+        return String(add(a, b));
+      case "-":
+        return String(subtract(a, b));
+      case "×":
+        return String(multiply(a, b));
+      case "^":
+        return String(power(a, b));
+      case "÷":
+        // Handle a division by zero
+        if (b === 0) {
+          return "Error";
+        }
+        return String(divide(a, b));
+    }
+  }
+
+  function add(a, b) {
```

Comment:
I don't think these functions are really needed, you can just use the operators inline.